src/Common/Utils.hs

```diff
@@ -47,4 +51,10 @@ uncurry5 f (a, b, c, d, e) = f a b c d e
 -- | Given a set of indices and a list, extracts the elements at those indices.
 -- Note that if the indices are out of bounds, this function will panic.
 getIndices :: [Int] -> [a] -> [a]
-getIndices indices xs = map (xs !!) indices
\ No newline at end of file
+getIndices indices xs = map (xs !!) indices
+
+allUnique :: (Ord a) => [a] -> Bool
+allUnique xs = length xs == length (fromList xs)
```

Comment:
In this code currently, `xs :: [a]`. So `fromList xs` is a no-op - did you intend this function to have type `(Ord a) => Set a -> Bool`?

Reply:
Oh, I'm a fool. `fromList` returns a `Set` disregard this one.